engine/runtime/src/main/java/org/enso/interpreter/node/expression/literal/IntegerLiteralNode.java

```diff
@@ -33,4 +36,13 @@ public static IntegerLiteralNode build(long value) {
   public Object executeGeneric(VirtualFrame frame) {
     return this.value;
   }
+
+  @Override
+  public Object parsePatch(IR.Expression ir) {
+    if (ir instanceof IR$Literal$Number n && n.numericValue() instanceof Long l) {
```

Comment:
just curious, is it possible this is an Int or Double?

Reply:
Not right now. `Double` would be represented by `DecimalLiteralNode`. I plan to merge all `LiteralNode` into one _"later"_.